### ISSUE TITLE
why not recommend KD alone as default recipe?

### ISSUE THREAD

**Issue by qianyizhang** [Alibaba-MIIL/Solving_ImageNet#4]
hi,
interesting paper.
based on Table.6, it seems better to use KD alone, by saving some ops on CE and ignoring the effort to tune hyper-parameter αkd.

Furthermore, I would be interested to see this training scheme im a mean-teacher setting, where the teacher model is not already trained, but rather a EMA of student. Have you tried that?

**Comment by qianyizhang** [comment]
> 1. KD-only gives the same results (within the statistical error) as KD+original labels. so you definitely can use only the KD teacher. This result is consistent with what we saw also in other experiments on downstream tasks - KD is such a powerful method, that it enables to remove the original labels, and use only the teacher predictions.
> 2. i have not tried using EMA of a student as a teacher. my strong guess is that it will require more training epochs, be more sensitive to hyper-params, will require stronger weight to the original labels, and will lead to somewhat worse results.
>    The reason - the most important phase of a training is the beginning, and in the beginning, the EMA teacher does not have valid predictions.

thanks for the fast and honest reply.
my main question in regard #1 is not about the validity of KD, but rather why not KD alone as your main recommendation, since your default recipe sets αkd = 5.

**Comment by mrT23** [comment]
KD alone is a rather risky option - you fully ignore the labels, treat the dataset as an unlabeled one, and trust only the teacher model to guide the student.
ImageNet score in a specific training is not the only thing that matters. maintaining a degree of freedom that still enable to utilize the original label is important in my opinion. who knows - maybe with some untested teacher, or in some other scenario, kd_only won't work well.

anyway, the code has an _--use_kd_only_loss_ flag. you are welcome to use it. from my tests, it does not hurt results.
